io.itch.doctorm64.skippybot.yaml

```diff
@@ -0,0 +1,44 @@
+app-id: io.itch.doctorm64.skippybot
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+tags:
+  - proprietary
+finish-args:
+- --socket=wayland
+- --socket=fallback-x11
+- --share=ipc
+
+# sound and graphics
+- --socket=pulseaudio
+- --device=dri
+
+command: skippybot
+modules:
+  - name: skippybot
+    build-options:
+      env:
+        SKIPPY_VERSION: "115"
+    buildsystem: simple
+    build-commands:
+      # Game
+      - install -Dm644 UnityPlayer.so /app/bin/UnityPlayer.so
+      - install -Dm755 skippy_${SKIPPY_VERSION}.x86_64 /app/bin/skippybot
+      - cp -r skippy_${SKIPPY_VERSION}_Data/ /app/bin/skippybot_Data
+      # meta files
+      - install -Dm644 $FLATPAK_ID.appdata.xml /app/share/metainfo/$FLATPAK_ID.appdata.xml
+      - install -Dm644 $FLATPAK_ID.png /app/share/icons/hicolor/128x128/apps/$FLATPAK_ID.png
+      - install -Dm644 $FLATPAK_ID.desktop /app/share/applications/$FLATPAK_ID.desktop
+    sources:
+      - type: archive
+        # redirected URL because itch.io does not allow direct downloads of assets. We are checking the hash tho, so file is the same.
```

Comment:
this basically will break on a whim.

also is redistribution authorized?

Reply:
Redistribution is authorized.

Can you go a bit more detail with how it'll break? 
I do find it quite sad that itch.io does not give a way to directly download assets, which leaves us with two options (to my knowledge at least):
- Use the redirected URL and update that on new versions and hope that the service itself never breaks
- Push the zip directly into the repository, and use it as a "file" source.

I personally opted for the former as having a 30MB binary commit on each new version is very ugly and bloats the repo too much IMO. However, if you think that that's a better option than relying on the third party, I'll use that instead.
And if you've got any other ideas on how to get around the issue, I'm all ears for that as well.

Reply:
if the sha256 is different that mean the build will be broken. 

A .zip in the flathub repo would be REJECTED.

Reply:
> if the sha256 is different that mean the build will be broken.

*If* the hash is different the build will be broken, yes, but the hash should never be different (and it never was in my few months of using it).

Reply:
Since redistribution of the assets are allowed, one could technically repost the asset to some other place (gdrive, owncloud etc.) and then use that as a source.
Although that probably wouldn't be that different from the current approach.

Reply:
if the url provide the same content in a stable way (and get changed on a new release) then to shouldn't be a problem. I initially understood that it wasn't like that.

Reply:
I'm sorry then if I made that unclear initially.
But yeah, just for clarification again, the current matix link is just a redirect to the actual asset on itch.io in order to directly download the asset there.
I'll try to make the comment a bit more clear.